// sanity/lib/_chunks-es/VariantsTool.js
import { jsx, jsxs } from "react/jsx-runtime";
import { c } from "react/compiler-runtime";
import { useRouter, StateLink } from "sanity/router";
import { Autocomplete, Stack, Flex, Box, Text, TextInput, TextArea, useToast, Card, Menu, Container, rem, Skeleton } from "@sanity/ui";
import { useId, useState, memo, forwardRef } from "react";
import { Button, Dialog, MenuButton, MenuItem } from "./index.js";
import { getVariantId, useClient, VARIANTS_STUDIO_CLIENT_OPTIONS, VARIANT_DOCUMENTS_PATH, useTranslation, variantsLocaleNamespace, useAllVariants, TextWithTone, ContextMenuButton, RelativeTime, useSchema, SanityDefaultPreview, decodeVariantIdFromRoute, LoadingBlock, getVariantDescription, getVariantConditionsText, getVariantTitle, filterVariantsForSearch } from "./index2.js";
import "history";
import "sanity/_singletons";
import "rxjs";
import "rxjs/operators";
import "react-rx";
import "@sanity/client/csm";
import "quick-lru";
import "lodash-es/throttle.js";
import "lodash-es/isEqual.js";
import { TrashIcon, AddIcon, DiamondIcon, SearchIcon } from "@sanity/icons";
import "@sanity/uuid";
import "@sanity/color";
import "color2k";
import "styled-components";
import "lodash-es/isString.js";
import "react-i18next";
import "./createAgentBundlesStore.js";
import "@sanity/telemetry/react";
import "@portabletext/editor";
import "@sanity/util/paths";
import "@sanity/types";
import "@sanity/diff-match-patch";
import "lodash-es/flatten.js";
import "lodash-es/orderBy.js";
import "lodash-es/uniq.js";
import "lodash-es/xor.js";
import "@sanity/diff-patch";
import "./version.js";
import "@sanity/ui/theme";
import "date-fns/differenceInDays";
import "date-fns/differenceInHours";
import "date-fns/differenceInMinutes";
import "date-fns/differenceInMonths";
import "date-fns/differenceInSeconds";
import "date-fns/differenceInWeeks";
import "date-fns/differenceInYears";
import "@sanity/diff";
import "mendoza";
import "@portabletext/sanity-bridge";
import "@sanity/id-utils";
import "date-fns/addHours";
import "date-fns/startOfHour";
import "date-fns/format";
import "date-fns/isValid";
import "date-fns/parse";
import "@sanity/util/legacyDateFormat";
import "react-fast-compare";
import "motion/react";
import "lodash-es/pick.js";
import "lodash-es/get.js";
import "@algorithm.ts/lcs";
import "lodash-es/intersection.js";
import "json-stable-stringify";
import "shallow-equals";
import "scroll-into-view-if-needed";
import "@sanity/schema/_internal";
import "lodash-es/groupBy.js";
import "lodash-es/sortBy.js";
import "lodash-es/compact.js";
import "lodash-es/keyBy.js";
import "lodash-es/partition.js";
import "lodash-es/toLower.js";
import "lodash-es/union.js";
import "lodash-es/words.js";
import "lodash-es/isFinite.js";
import "observable-callback";
import { randomKey } from "@sanity/util/content";
import "lodash-es/isPlainObject.js";
import "@sanity/insert-menu";
import "@tanstack/react-virtual";
import "lodash-es/startCase.js";
import "speakingurl";
import "@sanity/asset-utils";
import "@sanity/image-url";
import { customAlphabet } from "nanoid";
import "lodash-es/castArray.js";
import "rxjs-mergemap-array";
import "@sanity/mutator";
import "@portabletext/react";
import "lodash-es/findIndex.js";
import "@xstate/react";
import "xstate";
import "@sentry/react";
import "@sanity/schema";
import "@sanity/util/client";
import "@sanity/util/concurrency-limiter";
import "lodash-es/uniqBy.js";
import "json-reduce";
import "rxjs-exhaustmap-with-trailing";
import "date-fns/isBefore";
import "date-fns/startOfMinute";
import "lodash-es/debounce.js";
import "react-focus-lock";
import "polished";
import "@sanity/client";
import "react-is";
import "exif-component";
import "lodash-es/pickBy.js";
import "use-hot-module-reload";
import "lodash-es/memoize.js";
import "lodash-es/sortedIndex.js";
import "lodash-es/difference.js";
import "lodash-es/identity.js";
import "lodash-es/values.js";
import "@sanity/comlink";
import "lodash-es/omit.js";
import "@isaacs/ttlcache";
import "dataloader";
import "raf";
import "groq-js";
import "use-sync-external-store/with-selector";
import "lodash-es/escapeRegExp.js";
import "lodash-es/isEmpty.js";
import "react-dom/server";
import "lodash-es/capitalize.js";
import "lodash-es/last.js";
import "react-dom/client";
import "date-fns/addWeeks";
import "date-fns/isAfter";
import "semver";
import "lodash-es/isObject.js";
import { isPortableTextBlock, toPlainText } from "@portabletext/toolkit";
import { at, set } from "@sanity/mutate";
import { applyPatches } from "@sanity/mutate/_unstable_apply";
import { Table, Headers } from "./Table.js";
function createVariantOperationsStore(options) {
  const client = options.client;
  return {
    createVariant: async (variant) => {
      const action = {
        actionType: "sanity.action.variant.definition.create",
        variantId: getVariantId(variant._id),
        conditions: variant.conditions,
        priority: variant.priority,
        ...variant.metadata ? {
          metadata: variant.metadata
        } : {}
      };
      return await client.action(action, {
        tag: "variants.create"
      });
    },
    updateVariant: async (variant) => {
      const variantId = getVariantId(variant._id), setPayload = {
        conditions: variant.conditions,
        priority: variant.priority
      };
      variant.metadata && (setPayload.metadata = variant.metadata);
      const patch = {
        set: setPayload
      };
      variant.metadata || (patch.unset = ["metadata"]);
      const action = {
        actionType: "sanity.action.variant.definition.edit",
        variantId,
        patch
      };
      return await client.action(action, {
        tag: "variants.edit"
      });
    },
    deleteVariant: async (variantIdOrDocumentId) => {
      const action = {
        actionType: "sanity.action.variant.definition.delete",
        variantId: getVariantId(variantIdOrDocumentId)
      };
      return await client.action(action, {
        tag: "variants.delete"
      });
    }
  };
}
function useVariantOperations() {
  const $ = c(2), studioClient = useClient(VARIANTS_STUDIO_CLIENT_OPTIONS);
  let t0;
  return $[0] !== studioClient ? (t0 = createVariantOperationsStore({
    client: studioClient
  }), $[0] = studioClient, $[1] = t0) : t0 = $[1], t0;
}
function getVariantTitleValue(variant) {
  const title = variant.metadata?.title;
  return typeof title == "string" ? title.trim() : "";
}
function getIsVariantInvalid(variant) {
  return !getVariantTitleValue(variant) || !getHasValidConditions(variant.conditions);
}
function getHasValidConditions(conditions) {
  const entries = Object.entries(conditions);
  return entries.length > 0 && entries.every(([key, value]) => !!(key.trim() && value.trim()));
}
const CONDITION_KEY_REGEX = /^[a-z][a-z0-9_-]{0,63}$/, RESERVED_KEY_PREFIXES = ["_", "$"];
function getConditionKeyValidationError(key) {
  const trimmedKey = key.trim();
  if (trimmedKey) {
    for (const prefix of RESERVED_KEY_PREFIXES)
      if (trimmedKey.startsWith(prefix))
        return "reserved";
    if (trimmedKey.includes(":") || !CONDITION_KEY_REGEX.test(trimmedKey))
      return "invalid";
  }
}
function getConditionValueValidationError(value) {
  if (!value.trim())
    return "empty";
  if (value.includes(":"))
    return "invalid";
}
const createVariantIdSuffix = customAlphabet("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789", 8);
function createVariantId() {
  return `${VARIANT_DOCUMENTS_PATH}.${createVariantIdSuffix()}`;
}
function getVariantDefaults() {
  return {
    _id: createVariantId(),
    _type: "system.variant",
    conditions: {},
    priority: 0,
    metadata: {
      title: "",
      description: []
    }
  };
}
function createPortableTextDescription(text) {
  return text.trim() ? [{
    _key: randomKey(12),
    _type: "block",
    children: [{
      _key: randomKey(12),
      _type: "span",
      marks: [],
      text
    }],
    markDefs: [],
    style: "normal"
  }] : [];
}
function sortValues(values) {
  return Array.from(values).filter(Boolean).toSorted((a, b) => a.localeCompare(b));
}
function buildConditionSuggestionIndex(variants) {
  const keys = /* @__PURE__ */ new Set(), valuesByKey = /* @__PURE__ */ new Map();
  return variants.forEach((variant) => {
    Object.entries(variant.conditions).forEach(([rawKey, rawValue]) => {
      const key = rawKey.trim();
      if (!key)
        return;
      keys.add(key);
      const value = rawValue?.trim();
      if (!value)
        return;
      let values = valuesByKey.get(key);
      values || (values = /* @__PURE__ */ new Set(), valuesByKey.set(key, values)), values.add(value);
    });
  }), {
    keys: sortValues(keys),
    valuesByKey: new Map(Array.from(valuesByKey, ([key, values]) => [key, sortValues(values)]))
  };
}
function getConditionKeyOptions(index, rows, currentRowIndex) {
  const usedKeys = new Set(rows.filter((_, rowIndex) => rowIndex !== currentRowIndex).map((row) => row.key.trim()).filter(Boolean));
  return index.keys.filter((key) => !usedKeys.has(key)).map((value) => ({
    value
  }));
}
function getConditionValueOptions(index, conditionKey) {
  const key = conditionKey.trim();
  return key ? index.valuesByKey.get(key)?.map((value) => ({
    value
  })) ?? [] : [];
}
function filterConditionOption(query, option) {
  return option.value.toLowerCase().includes(query.trim().toLowerCase());
}
function ConditionAutocompleteInput(props) {
  const $ = c(16), {
    ariaLabel,
    autoFocus,
    customValidity,
    invalid,
    onChange,
    options,
    placeholder,
    testId,
    value
  } = props, id = useId(), [focused, setFocused] = useState(!1);
  let t0;
  $[0] !== onChange ? (t0 = (nextValue) => {
    nextValue !== null && onChange(nextValue);
  }, $[0] = onChange, $[1] = t0) : t0 = $[1];
  const handleQueryChange = t0, t1 = invalid ? "true" : void 0;
  let t2;
  $[2] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t2 = () => setFocused(!1), $[2] = t2) : t2 = $[2];
  let t3;
  $[3] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t3 = () => setFocused(!0), $[3] = t3) : t3 = $[3];
  const t4 = focused ? void 0 : value;
  let t5;
  return $[4] !== ariaLabel || $[5] !== autoFocus || $[6] !== customValidity || $[7] !== handleQueryChange || $[8] !== id || $[9] !== onChange || $[10] !== options || $[11] !== placeholder || $[12] !== t1 || $[13] !== t4 || $[14] !== testId ? (t5 = /* @__PURE__ */ jsx(Autocomplete, { id, "aria-invalid": t1, "aria-label": ariaLabel, autoFocus, customValidity, "data-testid": testId, filterOption: filterConditionOption, fontSize: 1, onBlur: t2, onChange, onFocus: t3, onQueryChange: handleQueryChange, openButton: !0, options, placeholder, value: t4 }), $[4] = ariaLabel, $[5] = autoFocus, $[6] = customValidity, $[7] = handleQueryChange, $[8] = id, $[9] = onChange, $[10] = options, $[11] = placeholder, $[12] = t1, $[13] = t4, $[14] = testId, $[15] = t5) : t5 = $[15], t5;
}
function getConditionRows(conditions) {
  const rows = Object.entries(conditions).map(([key, value]) => ({
    id: randomKey(12),
    key,
    value
  }));
  return rows.length ? rows : [{
    id: randomKey(12),
    key: "",
    value: ""
  }];
}
function getConditionsFromRows(rows) {
  return rows.reduce((conditions, row) => {
    const key = row.key.trim(), value = row.value.trim();
    return key && value && (conditions[key] = value), conditions;
  }, {});
}
function isConditionRowEmpty(row) {
  return !row.key.trim() && !row.value.trim();
}
function getEmptyConditionRowValidation() {
  return {
    key: null,
    value: null
  };
}
function getConditionRowsValidation(rows) {
  const conditionRowsValidation = /* @__PURE__ */ new Map(), seenKeys = /* @__PURE__ */ new Set();
  return rows.forEach((row, index) => {
    const validation = {
      key: null,
      value: null
    }, key = row.key.trim(), value = row.value.trim();
    if (key) {
      seenKeys.has(key) ? validation.key = "dialog.create.condition-key.duplicate" : seenKeys.add(key);
      const keyError = getConditionKeyValidationError(row.key);
      !validation.key && keyError === "reserved" ? validation.key = "dialog.create.condition-key.reserved" : !validation.key && keyError === "invalid" && (validation.key = "dialog.create.condition-key.invalid");
    } else value && (validation.key = "dialog.create.condition-key.required");
    const valueError = getConditionValueValidationError(row.value);
    (key || isConditionRowEmpty(row)) && valueError === "empty" ? validation.value = "dialog.create.condition-value.required" : valueError === "invalid" && (validation.value = "dialog.create.condition-value.invalid"), conditionRowsValidation.set(index, validation);
  }), conditionRowsValidation;
}
function hasConditionRowsValidationErrors(validation) {
  return Array.from(validation.values()).some(({
    key,
    value
  }) => key || value);
}
function getPortableTextDescriptionValue(description) {
  return !Array.isArray(description) || !description.every(isPortableTextBlock) ? "" : toPlainText(description);
}
function VariantForm(props) {
  const $ = c(108), {
    onChange,
    onConditionValidityChange,
    showValidation: t0,
    value
  } = props, showValidation = t0 === void 0 ? !1 : t0, {
    t
  } = useTranslation(variantsLocaleNamespace), {
    data: variants
  } = useAllVariants();
  let t1;
  $[0] !== variants ? (t1 = buildConditionSuggestionIndex(variants), $[0] = variants, $[1] = t1) : t1 = $[1];
  const suggestionIndex = t1, titleId = useId(), descriptionId = useId();
  let t2;
  $[2] !== value.conditions ? (t2 = () => getConditionRows(value.conditions), $[2] = value.conditions, $[3] = t2) : t2 = $[3];
  const [conditionRows, setConditionRows] = useState(t2);
  let t3;
  $[4] !== conditionRows ? (t3 = getConditionRowsValidation(conditionRows), $[4] = conditionRows, $[5] = t3) : t3 = $[5];
  const conditionsValidation = t3;
  let t4;
  $[6] !== value ? (t4 = getVariantTitleValue(value), $[6] = value, $[7] = t4) : t4 = $[7];
  const showTitleError = showValidation && !t4, lastConditionRow = conditionRows[conditionRows.length - 1];
  let t5;
  $[8] !== conditionsValidation || $[9] !== lastConditionRow ? (t5 = lastConditionRow && !hasConditionRowsValidationErrors(conditionsValidation), $[8] = conditionsValidation, $[9] = lastConditionRow, $[10] = t5) : t5 = $[10];
  const canAddCondition = !!t5;
  let t6;
  $[11] !== onChange || $[12] !== onConditionValidityChange ? (t6 = (nextRows) => {
    const rows = nextRows.length ? nextRows : getConditionRows({});
    setConditionRows(rows);
    const nextRowsValidation = getConditionRowsValidation(rows), nextRowsInvalid = hasConditionRowsValidationErrors(nextRowsValidation);
    onConditionValidityChange?.(nextRowsInvalid), (nextRows.length === 0 || !nextRowsInvalid) && onChange(["conditions"], getConditionsFromRows(nextRows));
  }, $[11] = onChange, $[12] = onConditionValidityChange, $[13] = t6) : t6 = $[13];
  const updateConditionRows = t6;
  let t7;
  $[14] !== onChange ? (t7 = (event) => {
    onChange(["metadata", "title"], event.currentTarget.value);
  }, $[14] = onChange, $[15] = t7) : t7 = $[15];
  const handleTitleChange = t7;
  let t8;
  $[16] !== onChange ? (t8 = (event_0) => {
    onChange(["metadata", "description"], createPortableTextDescription(event_0.currentTarget.value));
  }, $[16] = onChange, $[17] = t8) : t8 = $[17];
  const handleDescriptionChange = t8;
  let t9;
  $[18] !== conditionRows || $[19] !== updateConditionRows ? (t9 = (index, field, nextValue) => {
    const nextRows_0 = conditionRows.map((row, rowIndex) => rowIndex === index ? {
      ...row,
      [field]: nextValue
    } : row);
    updateConditionRows(nextRows_0);
  }, $[18] = conditionRows, $[19] = updateConditionRows, $[20] = t9) : t9 = $[20];
  const handleConditionChange = t9;
  let t10;
  $[21] !== canAddCondition || $[22] !== conditionRows || $[23] !== updateConditionRows ? (t10 = () => {
    canAddCondition && updateConditionRows([...conditionRows, {
      id: randomKey(12),
      key: "",
      value: ""
    }]);
  }, $[21] = canAddCondition, $[22] = conditionRows, $[23] = updateConditionRows, $[24] = t10) : t10 = $[24];
  const handleAddCondition = t10;
  let t11;
  $[25] !== conditionRows || $[26] !== updateConditionRows ? (t11 = (index_0) => {
    const nextRows_1 = conditionRows.filter((_, rowIndex_0) => rowIndex_0 !== index_0);
    updateConditionRows(nextRows_1);
  }, $[25] = conditionRows, $[26] = updateConditionRows, $[27] = t11) : t11 = $[27];
  const handleRemoveCondition = t11;
  let t12;
  $[28] !== t ? (t12 = t("dialog.create.variant-title.label"), $[28] = t, $[29] = t12) : t12 = $[29];
  let t13;
  $[30] !== t12 || $[31] !== titleId ? (t13 = /* @__PURE__ */ jsx(Text, { as: "label", htmlFor: titleId, size: 1, weight: "medium", children: t12 }), $[30] = t12, $[31] = titleId, $[32] = t13) : t13 = $[32];
  const t14 = showTitleError ? "true" : void 0;
  let t15;
  $[33] !== showTitleError || $[34] !== t ? (t15 = showTitleError ? t("dialog.create.variant-title.required") : void 0, $[33] = showTitleError, $[34] = t, $[35] = t15) : t15 = $[35];
  let t16;
  $[36] !== t ? (t16 = t("dialog.create.variant-title.placeholder"), $[36] = t, $[37] = t16) : t16 = $[37];
  const t17 = typeof value.metadata?.title == "string" ? value.metadata.title : "";
  let t18;
  $[38] !== handleTitleChange || $[39] !== t14 || $[40] !== t15 || $[41] !== t16 || $[42] !== t17 || $[43] !== titleId ? (t18 = /* @__PURE__ */ jsx(TextInput, { autoFocus: !0, "aria-invalid": t14, customValidity: t15, "data-testid": "variant-form-title", fontSize: 2, id: titleId, onChange: handleTitleChange, placeholder: t16, value: t17 }), $[38] = handleTitleChange, $[39] = t14, $[40] = t15, $[41] = t16, $[42] = t17, $[43] = titleId, $[44] = t18) : t18 = $[44];
  let t19;
  $[45] !== showTitleError || $[46] !== t ? (t19 = showTitleError && /* @__PURE__ */ jsx(TextWithTone, { "data-testid": "variant-form-title-error", size: 1, tone: "critical", children: t("dialog.create.variant-title.required") }), $[45] = showTitleError, $[46] = t, $[47] = t19) : t19 = $[47];
  let t20;
  $[48] !== t13 || $[49] !== t18 || $[50] !== t19 ? (t20 = /* @__PURE__ */ jsxs(Stack, { space: 3, children: [
    t13,
    t18,
    t19
  ] }), $[48] = t13, $[49] = t18, $[50] = t19, $[51] = t20) : t20 = $[51];
  let t21;
  $[52] !== t ? (t21 = t("dialog.create.description.label"), $[52] = t, $[53] = t21) : t21 = $[53];
  let t22;
  $[54] !== descriptionId || $[55] !== t21 ? (t22 = /* @__PURE__ */ jsx(Text, { as: "label", htmlFor: descriptionId, size: 1, weight: "medium", children: t21 }), $[54] = descriptionId, $[55] = t21, $[56] = t22) : t22 = $[56];
  let t23;
  $[57] !== t ? (t23 = t("dialog.create.description.placeholder"), $[57] = t, $[58] = t23) : t23 = $[58];
  const t24 = value.metadata?.description;
  let t25;
  $[59] !== t24 ? (t25 = getPortableTextDescriptionValue(t24), $[59] = t24, $[60] = t25) : t25 = $[60];
  let t26;
  $[61] !== descriptionId || $[62] !== handleDescriptionChange || $[63] !== t23 || $[64] !== t25 ? (t26 = /* @__PURE__ */ jsx(TextArea, { "data-testid": "variant-form-description", fontSize: 1, id: descriptionId, onChange: handleDescriptionChange, placeholder: t23, rows: 3, value: t25 }), $[61] = descriptionId, $[62] = handleDescriptionChange, $[63] = t23, $[64] = t25, $[65] = t26) : t26 = $[65];
  let t27;
  $[66] !== t22 || $[67] !== t26 ? (t27 = /* @__PURE__ */ jsxs(Stack, { space: 3, children: [
    t22,
    t26
  ] }), $[66] = t22, $[67] = t26, $[68] = t27) : t27 = $[68];
  let t28;
  $[69] !== t ? (t28 = t("dialog.create.conditions.title"), $[69] = t, $[70] = t28) : t28 = $[70];
  let t29;
  $[71] !== t28 ? (t29 = /* @__PURE__ */ jsx(Text, { size: 1, weight: "medium", children: t28 }), $[71] = t28, $[72] = t29) : t29 = $[72];
  let t30;
  $[73] !== t ? (t30 = t("dialog.create.conditions.description"), $[73] = t, $[74] = t30) : t30 = $[74];
  let t31;
  $[75] !== t30 ? (t31 = /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: t30 }), $[75] = t30, $[76] = t31) : t31 = $[76];
  let t32;
  $[77] !== t29 || $[78] !== t31 ? (t32 = /* @__PURE__ */ jsxs(Stack, { space: 2, children: [
    t29,
    t31
  ] }), $[77] = t29, $[78] = t31, $[79] = t32) : t32 = $[79];
  let t33;
  $[80] !== conditionRows || $[81] !== conditionsValidation || $[82] !== handleConditionChange || $[83] !== handleRemoveCondition || $[84] !== showValidation || $[85] !== suggestionIndex || $[86] !== t ? (t33 = conditionRows.map((row_0, index_1) => {
    const validation = conditionsValidation.get(index_1) ?? getEmptyConditionRowValidation(), valueValidation = showValidation ? validation.value : null, conditionValidationError = validation.key || valueValidation;
    return /* @__PURE__ */ jsxs(Stack, { space: 2, children: [
      /* @__PURE__ */ jsxs(Flex, { align: "center", gap: 2, children: [
        /* @__PURE__ */ jsx(Box, { flex: 1, children: /* @__PURE__ */ jsx(ConditionAutocompleteInput, { autoFocus: index_1 > 0, ariaLabel: t("dialog.create.condition-key.label"), customValidity: validation.key ? t(validation.key) : void 0, invalid: !!validation.key, onChange: (nextValue_0) => handleConditionChange(index_1, "key", nextValue_0), options: getConditionKeyOptions(suggestionIndex, conditionRows, index_1), placeholder: t("dialog.create.condition-key.placeholder"), testId: "variant-form-condition-key", value: row_0.key }) }),
        /* @__PURE__ */ jsx(Box, { flex: 1, children: /* @__PURE__ */ jsx(ConditionAutocompleteInput, { ariaLabel: t("dialog.create.condition-value.label"), customValidity: valueValidation ? t(valueValidation) : void 0, invalid: !!valueValidation, onChange: (nextValue_1) => handleConditionChange(index_1, "value", nextValue_1), options: getConditionValueOptions(suggestionIndex, row_0.key), placeholder: t("dialog.create.condition-value.placeholder"), testId: "variant-form-condition-value", value: row_0.value }) }),
        /* @__PURE__ */ jsx(Button, { disabled: isConditionRowEmpty(row_0) && conditionRows.length === 1, icon: TrashIcon, mode: "bleed", onClick: () => handleRemoveCondition(index_1), tone: "critical", tooltipProps: {
          content: t("dialog.create.remove-condition")
        }, type: "button" })
      ] }),
      conditionValidationError && /* @__PURE__ */ jsx(TextWithTone, { "data-testid": validation.key ? "variant-form-condition-key-error" : "variant-form-condition-value-error", size: 1, tone: "critical", children: t(conditionValidationError) })
    ] }, row_0.id);
  }), $[80] = conditionRows, $[81] = conditionsValidation, $[82] = handleConditionChange, $[83] = handleRemoveCondition, $[84] = showValidation, $[85] = suggestionIndex, $[86] = t, $[87] = t33) : t33 = $[87];
  let t34;
  $[88] !== t33 ? (t34 = /* @__PURE__ */ jsx(Stack, { space: 2, children: t33 }), $[88] = t33, $[89] = t34) : t34 = $[89];
  const t35 = !canAddCondition;
  let t36;
  $[90] !== t ? (t36 = t("dialog.create.action.add-condition"), $[90] = t, $[91] = t36) : t36 = $[91];
  let t37;
  $[92] !== canAddCondition || $[93] !== t ? (t37 = canAddCondition ? null : {
    content: t("dialog.create.action.add-condition.disabled-hint")
  }, $[92] = canAddCondition, $[93] = t, $[94] = t37) : t37 = $[94];
  let t38;
  $[95] !== handleAddCondition || $[96] !== t35 || $[97] !== t36 || $[98] !== t37 ? (t38 = /* @__PURE__ */ jsx(Flex, { children: /* @__PURE__ */ jsx(Button, { disabled: t35, icon: AddIcon, mode: "ghost", onClick: handleAddCondition, text: t36, tooltipProps: t37, type: "button" }) }), $[95] = handleAddCondition, $[96] = t35, $[97] = t36, $[98] = t37, $[99] = t38) : t38 = $[99];
  let t39;
  $[100] !== t32 || $[101] !== t34 || $[102] !== t38 ? (t39 = /* @__PURE__ */ jsxs(Stack, { space: 3, children: [
    t32,
    t34,
    t38
  ] }), $[100] = t32, $[101] = t34, $[102] = t38, $[103] = t39) : t39 = $[103];
  let t40;
  return $[104] !== t20 || $[105] !== t27 || $[106] !== t39 ? (t40 = /* @__PURE__ */ jsxs(Stack, { space: 5, children: [
    t20,
    t27,
    t39
  ] }), $[104] = t20, $[105] = t27, $[106] = t39, $[107] = t40) : t40 = $[107], t40;
}
function VariantDialog(props) {
  const $ = c(35), {
    confirmDataTestId,
    confirmText,
    errorTitle,
    header,
    id,
    initialValue,
    onCancel,
    onSubmit,
    renderCancelButton: t0
  } = props, renderCancelButton = t0 === void 0 ? !1 : t0, toast = useToast(), {
    t
  } = useTranslation(variantsLocaleNamespace), [variant, setVariant] = useState(initialValue), [isSubmitting, setIsSubmitting] = useState(!1), [showValidation, setShowValidation] = useState(!1), [conditionsInvalid, setConditionsInvalid] = useState(!1);
  let t1;
  $[0] !== conditionsInvalid || $[1] !== variant ? (t1 = getIsVariantInvalid(variant) || conditionsInvalid, $[0] = conditionsInvalid, $[1] = variant, $[2] = t1) : t1 = $[2];
  const invalid = t1;
  let t2;
  $[3] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t2 = (path, nextValue) => {
    setVariant((currentVariant) => applyPatches([at(path, set(nextValue))], currentVariant));
  }, $[3] = t2) : t2 = $[3];
  const handleVariantChange = t2;
  let t3;
  $[4] !== errorTitle || $[5] !== invalid || $[6] !== onSubmit || $[7] !== toast || $[8] !== variant ? (t3 = async (event) => {
    if (event.preventDefault(), invalid) {
      setShowValidation(!0);
      return;
    }
    setIsSubmitting(!0);
    try {
      await onSubmit(variant), setIsSubmitting(!1);
    } catch (t42) {
      const error = t42;
      setIsSubmitting(!1), console.error(error), toast.push({
        closable: !0,
        status: "error",
        title: errorTitle
      });
    }
  }, $[4] = errorTitle, $[5] = invalid, $[6] = onSubmit, $[7] = toast, $[8] = variant, $[9] = t3) : t3 = $[9];
  const handleSubmit = t3, t4 = isSubmitting ? void 0 : onCancel, t5 = isSubmitting ? void 0 : onCancel;
  let t6;
  $[10] !== showValidation || $[11] !== variant ? (t6 = /* @__PURE__ */ jsx(Box, { paddingBottom: 4, children: /* @__PURE__ */ jsx(VariantForm, { onChange: handleVariantChange, onConditionValidityChange: setConditionsInvalid, showValidation, value: variant }) }), $[10] = showValidation, $[11] = variant, $[12] = t6) : t6 = $[12];
  let t7;
  $[13] !== isSubmitting || $[14] !== onCancel || $[15] !== renderCancelButton || $[16] !== t ? (t7 = renderCancelButton && /* @__PURE__ */ jsx(Button, { disabled: isSubmitting, mode: "ghost", onClick: onCancel, text: t("dialog.edit.action.cancel"), type: "button" }), $[13] = isSubmitting, $[14] = onCancel, $[15] = renderCancelButton, $[16] = t, $[17] = t7) : t7 = $[17];
  let t8;
  $[18] !== confirmDataTestId || $[19] !== confirmText || $[20] !== isSubmitting ? (t8 = /* @__PURE__ */ jsx(Button, { "data-testid": confirmDataTestId, disabled: isSubmitting, loading: isSubmitting, size: "large", text: confirmText, type: "submit" }), $[18] = confirmDataTestId, $[19] = confirmText, $[20] = isSubmitting, $[21] = t8) : t8 = $[21];
  let t9;
  $[22] !== t7 || $[23] !== t8 ? (t9 = /* @__PURE__ */ jsxs(Flex, { gap: 2, justify: "flex-end", paddingTop: 5, children: [
    t7,
    t8
  ] }), $[22] = t7, $[23] = t8, $[24] = t9) : t9 = $[24];
  let t10;
  $[25] !== handleSubmit || $[26] !== t6 || $[27] !== t9 ? (t10 = /* @__PURE__ */ jsx(Card, { borderTop: !0, padding: 4, children: /* @__PURE__ */ jsxs("form", { onSubmit: handleSubmit, children: [
    t6,
    t9
  ] }) }), $[25] = handleSubmit, $[26] = t6, $[27] = t9, $[28] = t10) : t10 = $[28];
  let t11;
  return $[29] !== header || $[30] !== id || $[31] !== t10 || $[32] !== t4 || $[33] !== t5 ? (t11 = /* @__PURE__ */ jsx(Dialog, { __unstable_autoFocus: !1, header, id, onClickOutside: t4, onClose: t5, padding: !1, width: 1, children: t10 }), $[29] = header, $[30] = id, $[31] = t10, $[32] = t4, $[33] = t5, $[34] = t11) : t11 = $[34], t11;
}
function toEditableVariant(variant) {
  return {
    _id: variant._id,
    _type: variant._type,
    conditions: variant.conditions,
    priority: variant.priority,
    metadata: variant.metadata
  };
}
function EditVariantDialog(props) {
  const $ = c(18), {
    onCancel,
    onSubmit,
    variant: initialVariant
  } = props, {
    t
  } = useTranslation(variantsLocaleNamespace), {
    updateVariant
  } = useVariantOperations();
  let t0;
  $[0] !== initialVariant ? (t0 = toEditableVariant(initialVariant), $[0] = initialVariant, $[1] = t0) : t0 = $[1];
  const initialValue = t0;
  let t1;
  $[2] !== onSubmit || $[3] !== updateVariant ? (t1 = async (variant) => {
    await updateVariant(variant), onSubmit();
  }, $[2] = onSubmit, $[3] = updateVariant, $[4] = t1) : t1 = $[4];
  const handleSubmit = t1;
  let t2;
  $[5] !== t ? (t2 = t("dialog.edit.action.confirm"), $[5] = t, $[6] = t2) : t2 = $[6];
  let t3;
  $[7] !== t ? (t3 = t("dialog.edit.error.title"), $[7] = t, $[8] = t3) : t3 = $[8];
  let t4;
  $[9] !== t ? (t4 = t("dialog.edit.title"), $[9] = t, $[10] = t4) : t4 = $[10];
  let t5;
  return $[11] !== handleSubmit || $[12] !== initialValue || $[13] !== onCancel || $[14] !== t2 || $[15] !== t3 || $[16] !== t4 ? (t5 = /* @__PURE__ */ jsx(VariantDialog, { confirmDataTestId: "save-variant-button", confirmText: t2, errorTitle: t3, header: t4, id: "edit-variant-dialog", initialValue, onCancel, onSubmit: handleSubmit, renderCancelButton: !0 }), $[11] = handleSubmit, $[12] = initialValue, $[13] = onCancel, $[14] = t2, $[15] = t3, $[16] = t4, $[17] = t5) : t5 = $[17], t5;
}
function VariantDetailMenuButton(t0) {
  const $ = c(21), {
    variant
  } = t0, {
    t
  } = useTranslation(variantsLocaleNamespace), router = useRouter(), toast = useToast(), {
    deleteVariant
  } = useVariantOperations(), [isDeleting, setIsDeleting] = useState(!1);
  let t1;
  $[0] !== deleteVariant || $[1] !== router || $[2] !== t || $[3] !== toast || $[4] !== variant._id ? (t1 = async () => {
    setIsDeleting(!0);
    try {
      await deleteVariant(variant._id), setIsDeleting(!1), router.navigate({});
    } catch (t22) {
      console.error(t22), toast.push({
        closable: !0,
        status: "error",
        title: t("overview.action.delete-variant.error.title")
      }), setIsDeleting(!1);
    }
  }, $[0] = deleteVariant, $[1] = router, $[2] = t, $[3] = toast, $[4] = variant._id, $[5] = t1) : t1 = $[5];
  const handleDelete = t1;
  let t2;
  $[6] !== isDeleting ? (t2 = /* @__PURE__ */ jsx(ContextMenuButton, { disabled: isDeleting, loading: isDeleting }), $[6] = isDeleting, $[7] = t2) : t2 = $[7];
  let t3;
  $[8] !== variant._id ? (t3 = getVariantId(variant._id), $[8] = variant._id, $[9] = t3) : t3 = $[9];
  const t4 = `variant-detail-actions-${t3}`;
  let t5;
  $[10] !== t ? (t5 = t("overview.action.delete-variant"), $[10] = t, $[11] = t5) : t5 = $[11];
  let t6;
  $[12] !== handleDelete || $[13] !== isDeleting || $[14] !== t5 ? (t6 = /* @__PURE__ */ jsx(Menu, { children: /* @__PURE__ */ jsx(MenuItem, { disabled: isDeleting, onClick: handleDelete, text: t5, tone: "critical" }) }), $[12] = handleDelete, $[13] = isDeleting, $[14] = t5, $[15] = t6) : t6 = $[15];
  let t7;
  $[16] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t7 = {
    placement: "bottom-end",
    portal: !0
  }, $[16] = t7) : t7 = $[16];
  let t8;
  return $[17] !== t2 || $[18] !== t4 || $[19] !== t6 ? (t8 = /* @__PURE__ */ jsx(MenuButton, { button: t2, id: t4, menu: t6, popover: t7 }), $[17] = t2, $[18] = t4, $[19] = t6, $[20] = t8) : t8 = $[20], t8;
}
function VariantDetailFooter(t0) {
  const $ = c(14), {
    variant
  } = t0, {
    t
  } = useTranslation(variantsLocaleNamespace);
  let t1;
  $[0] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t1 = /* @__PURE__ */ jsx(Card, { borderTop: !0, marginX: 2, style: {
    opacity: 0.6
  } }), $[0] = t1) : t1 = $[0];
  let t2;
  $[1] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t2 = /* @__PURE__ */ jsx(Text, { size: 1, children: /* @__PURE__ */ jsx(DiamondIcon, {}) }), $[1] = t2) : t2 = $[1];
  let t3;
  $[2] !== t ? (t3 = t("detail.footer.created"), $[2] = t, $[3] = t3) : t3 = $[3];
  let t4;
  $[4] !== variant._createdAt ? (t4 = /* @__PURE__ */ jsx(RelativeTime, { time: variant._createdAt, useTemporalPhrase: !0, minimal: !0 }), $[4] = variant._createdAt, $[5] = t4) : t4 = $[5];
  let t5;
  $[6] !== t3 || $[7] !== t4 ? (t5 = /* @__PURE__ */ jsx(Flex, { flex: 1, gap: 1, children: /* @__PURE__ */ jsx(Card, { children: /* @__PURE__ */ jsxs(Flex, { align: "center", gap: 2, children: [
    t2,
    /* @__PURE__ */ jsxs(Text, { muted: !0, size: 1, children: [
      t3,
      " ",
      t4
    ] })
  ] }) }) }), $[6] = t3, $[7] = t4, $[8] = t5) : t5 = $[8];
  let t6;
  $[9] !== variant ? (t6 = /* @__PURE__ */ jsx(Flex, { flex: "none", gap: 1, "data-testid": "variant-detail-footer-actions", children: /* @__PURE__ */ jsx(VariantDetailMenuButton, { variant }) }), $[9] = variant, $[10] = t6) : t6 = $[10];
  let t7;
  return $[11] !== t5 || $[12] !== t6 ? (t7 = /* @__PURE__ */ jsxs(Card, { flex: "none", children: [
    t1,
    /* @__PURE__ */ jsxs(Flex, { padding: 3, children: [
      t5,
      t6
    ] })
  ] }), $[11] = t5, $[12] = t6, $[13] = t7) : t7 = $[13], t7;
}
const TABLE_CARD_STYLE = {
  height: "100%",
  overflow: "auto"
};
function getDocumentPreviewTitle(document) {
  const title = document.title || document.name;
  return typeof title == "string" && title.trim() ? title : document._id;
}
const MemoDocumentType = memo(function(t0) {
  const $ = c(5), {
    type
  } = t0, schema = useSchema();
  let t1;
  $[0] !== schema || $[1] !== type ? (t1 = schema.get(type), $[0] = schema, $[1] = type, $[2] = t1) : t1 = $[2];
  const t2 = t1?.title || type;
  let t3;
  return $[3] !== t2 ? (t3 = /* @__PURE__ */ jsx(Text, { size: 1, children: t2 }), $[3] = t2, $[4] = t3) : t3 = $[4], t3;
}, (prev, next) => prev.type === next.type);
function getVariantDocumentColumnDefs(t) {
  return [{
    id: "version",
    width: null,
    style: {
      minWidth: 100
    },
    sorting: !1,
    header: (props) => /* @__PURE__ */ jsx(Flex, { ...props.headerProps, paddingY: 3, sizing: "border", children: /* @__PURE__ */ jsx(Headers.BasicHeader, { text: t("detail.documents.table.version") }) }),
    cell: ({
      cellProps
    }) => /* @__PURE__ */ jsx(Flex, { align: "center", ...cellProps, children: /* @__PURE__ */ jsx(Box, { paddingX: 2, children: /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: "-" }) }) })
  }, {
    id: "document._type",
    width: null,
    style: {
      minWidth: 100
    },
    sorting: !0,
    header: (props) => /* @__PURE__ */ jsx(Flex, { ...props.headerProps, paddingY: 3, sizing: "border", children: /* @__PURE__ */ jsx(Headers.SortHeaderButton, { text: t("detail.documents.table.type"), ...props }) }),
    cell: ({
      cellProps,
      datum
    }) => /* @__PURE__ */ jsx(Flex, { align: "center", ...cellProps, children: /* @__PURE__ */ jsx(Box, { paddingX: 2, children: !datum.isLoading && /* @__PURE__ */ jsx(MemoDocumentType, { type: datum.document._type }) }) })
  }, {
    id: "search",
    width: null,
    style: {
      minWidth: "min(50%, calc(100vw - 80px))",
      maxWidth: "min(50%, calc(100vw - 80px))"
    },
    sortTransform: ({
      document
    }) => getDocumentPreviewTitle(document).toLowerCase(),
    header: (props) => /* @__PURE__ */ jsx(Headers.TableHeaderSearch, { ...props, placeholder: t("detail.documents.table.search-placeholder") }),
    cell: ({
      cellProps,
      datum
    }) => /* @__PURE__ */ jsx(Box, { ...cellProps, flex: 1, padding: 1, paddingRight: 2, sizing: "border", children: datum.isLoading ? /* @__PURE__ */ jsx(SanityDefaultPreview, { isPlaceholder: !0 }) : /* @__PURE__ */ jsx(SanityDefaultPreview, { title: getDocumentPreviewTitle(datum.document), subtitle: datum.document._id }) })
  }, {
    id: "document._updatedAt",
    sorting: !0,
    width: 130,
    header: (props) => /* @__PURE__ */ jsx(Flex, { ...props.headerProps, paddingY: 3, sizing: "border", children: /* @__PURE__ */ jsx(Headers.SortHeaderButton, { text: t("detail.documents.table.edited"), ...props }) }),
    cell: ({
      cellProps,
      datum
    }) => /* @__PURE__ */ jsx(Flex, { ...cellProps, align: "center", paddingX: 2, paddingY: 3, style: {
      minWidth: 130
    }, children: !datum.isLoading && datum.document._updatedAt && /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: /* @__PURE__ */ jsx(RelativeTime, { time: datum.document._updatedAt, useTemporalPhrase: !0, minimal: !0 }) }) })
  }];
}
function filterDocuments(rows, searchTerm) {
  const normalizedSearchTerm = searchTerm.trim().toLowerCase();
  return normalizedSearchTerm ? rows.filter(({
    document
  }) => [getDocumentPreviewTitle(document), document._id, document._type].some((value) => value.toLowerCase().includes(normalizedSearchTerm))) : rows;
}
function VariantDocumentsTable(t0) {
  const $ = c(12), {
    documents
  } = t0, {
    t
  } = useTranslation(variantsLocaleNamespace), [scrollContainerRef, setScrollContainerRef] = useState(null);
  let t1;
  $[0] !== t ? (t1 = getVariantDocumentColumnDefs(t), $[0] = t, $[1] = t1) : t1 = $[1];
  const columnDefs = t1;
  let t2;
  $[2] !== documents ? (t2 = documents.map(_temp$2), $[2] = documents, $[3] = t2) : t2 = $[3];
  const rows = t2;
  let t3;
  $[4] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t3 = {
    column: "search",
    direction: "asc"
  }, $[4] = t3) : t3 = $[4];
  let t4;
  $[5] !== t ? (t4 = t("detail.documents.no-documents"), $[5] = t, $[6] = t4) : t4 = $[6];
  let t5;
  return $[7] !== columnDefs || $[8] !== rows || $[9] !== scrollContainerRef || $[10] !== t4 ? (t5 = /* @__PURE__ */ jsx(Card, { flex: 1, ref: setScrollContainerRef, style: TABLE_CARD_STYLE, children: /* @__PURE__ */ jsx(Table, { columnDefs, data: rows, defaultSort: t3, emptyState: t4, rowId: "document._id", scrollContainerRef, searchFilter: filterDocuments }) }), $[7] = columnDefs, $[8] = rows, $[9] = scrollContainerRef, $[10] = t4, $[11] = t5) : t5 = $[11], t5;
}
function _temp$2(document) {
  return {
    document
  };
}
const EMPTY_VARIANT_DOCUMENTS = [];
function VariantDetail() {
  const $ = c(92), router = useRouter(), {
    t
  } = useTranslation(variantsLocaleNamespace), [editDialogOpen, setEditDialogOpen] = useState(!1), variantIdRaw = typeof router.state.variantId == "string" ? router.state.variantId : void 0, variantId = decodeVariantIdFromRoute(variantIdRaw), {
    byId,
    loading
  } = useAllVariants(), variant = variantId ? byId.get(variantId) : void 0;
  if (loading) {
    let t02;
    $[0] !== t ? (t02 = t("detail.loading"), $[0] = t, $[1] = t02) : t02 = $[1];
    let t110;
    return $[2] !== t02 ? (t110 = /* @__PURE__ */ jsx(LoadingBlock, { fill: !0, title: t02 }), $[2] = t02, $[3] = t110) : t110 = $[3], t110;
  }
  if (!variant) {
    let t02;
    $[4] !== router ? (t02 = () => router.navigate({}), $[4] = router, $[5] = t02) : t02 = $[5];
    let t110;
    $[6] !== t ? (t110 = t("detail.back"), $[6] = t, $[7] = t110) : t110 = $[7];
    let t210;
    $[8] !== t02 || $[9] !== t110 ? (t210 = /* @__PURE__ */ jsx(Card, { borderBottom: !0, flex: "none", padding: 3, children: /* @__PURE__ */ jsx(Button, { mode: "bleed", onClick: t02, text: t110 }) }), $[8] = t02, $[9] = t110, $[10] = t210) : t210 = $[10];
    let t310;
    $[11] !== t ? (t310 = t("detail.not-found.title"), $[11] = t, $[12] = t310) : t310 = $[12];
    let t46;
    $[13] !== t310 ? (t46 = /* @__PURE__ */ jsx(Text, { size: 2, weight: "semibold", children: t310 }), $[13] = t310, $[14] = t46) : t46 = $[14];
    let t52;
    $[15] !== t ? (t52 = t("detail.not-found.description"), $[15] = t, $[16] = t52) : t52 = $[16];
    let t62;
    $[17] !== t52 ? (t62 = /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: t52 }), $[17] = t52, $[18] = t62) : t62 = $[18];
    let t72;
    $[19] !== t46 || $[20] !== t62 ? (t72 = /* @__PURE__ */ jsx(Box, { padding: 4, children: /* @__PURE__ */ jsx(Card, { border: !0, padding: 4, radius: 3, children: /* @__PURE__ */ jsxs(Stack, { space: 3, children: [
      t46,
      t62
    ] }) }) }), $[19] = t46, $[20] = t62, $[21] = t72) : t72 = $[21];
    let t82;
    return $[22] !== t210 || $[23] !== t72 ? (t82 = /* @__PURE__ */ jsxs(Flex, { direction: "column", flex: 1, height: "fill", children: [
      t210,
      t72
    ] }), $[22] = t210, $[23] = t72, $[24] = t82) : t82 = $[24], t82;
  }
  const description = getVariantDescription(variant), conditionsText = getVariantConditionsText(variant.conditions), T0 = Flex, t0 = "column", t1 = 1, t2 = "fill", t3 = "hidden";
  let t4;
  $[25] !== router ? (t4 = () => router.navigate({}), $[25] = router, $[26] = t4) : t4 = $[26];
  let t5;
  $[27] !== t ? (t5 = t("detail.back"), $[27] = t, $[28] = t5) : t5 = $[28];
  let t6;
  $[29] !== t4 || $[30] !== t5 ? (t6 = /* @__PURE__ */ jsx(Card, { borderBottom: !0, flex: "none", padding: 3, children: /* @__PURE__ */ jsx(Button, { mode: "bleed", onClick: t4, text: t5 }) }), $[29] = t4, $[30] = t5, $[31] = t6) : t6 = $[31];
  const T1 = Flex, t7 = "column", t8 = 1, t9 = "fill", t10 = "hidden";
  let t11;
  $[32] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t11 = {
    minHeight: 0
  }, $[32] = t11) : t11 = $[32];
  const T2 = Container, t12 = "none", t13 = 3, T3 = Flex, t14 = "column", t15 = 3, T4 = Card, t16 = 5, T5 = Flex, t17 = "flex-start", t18 = 4, t19 = "space-between", T6 = Stack, t20 = 3, T7 = Text, t21 = "h1", t22 = 4, t23 = "bold", t24 = getVariantTitle(variant);
  let t25;
  $[33] !== T7 || $[34] !== t24 ? (t25 = /* @__PURE__ */ jsx(T7, { as: t21, size: t22, weight: t23, children: t24 }), $[33] = T7, $[34] = t24, $[35] = t25) : t25 = $[35];
  let t26;
  $[36] !== description || $[37] !== t ? (t26 = description || t("detail.no-description"), $[36] = description, $[37] = t, $[38] = t26) : t26 = $[38];
  let t27;
  $[39] !== t26 ? (t27 = /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: t26 }), $[39] = t26, $[40] = t27) : t27 = $[40];
  let t28;
  $[41] !== T6 || $[42] !== t25 || $[43] !== t27 ? (t28 = /* @__PURE__ */ jsxs(T6, { space: t20, children: [
    t25,
    t27
  ] }), $[41] = T6, $[42] = t25, $[43] = t27, $[44] = t28) : t28 = $[44];
  let t29;
  $[45] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t29 = () => setEditDialogOpen(!0), $[45] = t29) : t29 = $[45];
  let t30;
  $[46] !== t ? (t30 = t("detail.action.edit-variant"), $[46] = t, $[47] = t30) : t30 = $[47];
  let t31;
  $[48] !== t30 ? (t31 = /* @__PURE__ */ jsx(Button, { onClick: t29, text: t30 }), $[48] = t30, $[49] = t31) : t31 = $[49];
  let t32;
  $[50] !== T5 || $[51] !== t28 || $[52] !== t31 ? (t32 = /* @__PURE__ */ jsxs(T5, { align: t17, gap: t18, justify: t19, children: [
    t28,
    t31
  ] }), $[50] = T5, $[51] = t28, $[52] = t31, $[53] = t32) : t32 = $[53];
  let t33;
  $[54] !== t ? (t33 = t("dialog.create.conditions.title"), $[54] = t, $[55] = t33) : t33 = $[55];
  let t34;
  $[56] !== t33 ? (t34 = /* @__PURE__ */ jsx(Text, { size: 1, weight: "medium", children: t33 }), $[56] = t33, $[57] = t34) : t34 = $[57];
  let t35;
  $[58] !== conditionsText || $[59] !== t ? (t35 = conditionsText || t("overview.table.no-conditions"), $[58] = conditionsText, $[59] = t, $[60] = t35) : t35 = $[60];
  let t36;
  $[61] !== t35 ? (t36 = /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: t35 }), $[61] = t35, $[62] = t36) : t36 = $[62];
  let t37;
  $[63] !== t34 || $[64] !== t36 ? (t37 = /* @__PURE__ */ jsx(Box, { paddingTop: 4, children: /* @__PURE__ */ jsxs(Stack, { space: 2, children: [
    t34,
    t36
  ] }) }), $[63] = t34, $[64] = t36, $[65] = t37) : t37 = $[65];
  let t38;
  $[66] !== T4 || $[67] !== t32 || $[68] !== t37 ? (t38 = /* @__PURE__ */ jsxs(T4, { paddingY: t16, children: [
    t32,
    t37
  ] }), $[66] = T4, $[67] = t32, $[68] = t37, $[69] = t38) : t38 = $[69];
  let t39;
  $[70] !== T3 || $[71] !== t38 ? (t39 = /* @__PURE__ */ jsx(T3, { direction: t14, paddingX: t15, children: t38 }), $[70] = T3, $[71] = t38, $[72] = t39) : t39 = $[72];
  let t40;
  $[73] !== T2 || $[74] !== t39 ? (t40 = /* @__PURE__ */ jsx(T2, { flex: t12, width: t13, children: t39 }), $[73] = T2, $[74] = t39, $[75] = t40) : t40 = $[75];
  let t41;
  $[76] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t41 = /* @__PURE__ */ jsx(Flex, { direction: "column", flex: 1, overflow: "hidden", style: {
    minHeight: 0
  }, children: /* @__PURE__ */ jsx(VariantDocumentsTable, { documents: EMPTY_VARIANT_DOCUMENTS }) }), $[76] = t41) : t41 = $[76];
  let t42;
  $[77] !== T1 || $[78] !== t11 || $[79] !== t40 ? (t42 = /* @__PURE__ */ jsxs(T1, { direction: t7, flex: t8, height: t9, overflow: t10, style: t11, children: [
    t40,
    t41
  ] }), $[77] = T1, $[78] = t11, $[79] = t40, $[80] = t42) : t42 = $[80];
  let t43;
  $[81] !== variant ? (t43 = /* @__PURE__ */ jsx(VariantDetailFooter, { variant }), $[81] = variant, $[82] = t43) : t43 = $[82];
  let t44;
  $[83] !== editDialogOpen || $[84] !== variant ? (t44 = editDialogOpen && /* @__PURE__ */ jsx(EditVariantDialog, { onCancel: () => setEditDialogOpen(!1), onSubmit: () => setEditDialogOpen(!1), variant }), $[83] = editDialogOpen, $[84] = variant, $[85] = t44) : t44 = $[85];
  let t45;
  return $[86] !== T0 || $[87] !== t42 || $[88] !== t43 || $[89] !== t44 || $[90] !== t6 ? (t45 = /* @__PURE__ */ jsxs(T0, { direction: t0, flex: t1, height: t2, overflow: t3, children: [
    t6,
    t42,
    t43,
    t44
  ] }), $[86] = T0, $[87] = t42, $[88] = t43, $[89] = t44, $[90] = t6, $[91] = t45) : t45 = $[91], t45;
}
function CreateVariantDialog(props) {
  const $ = c(16), {
    onCancel,
    onSubmit
  } = props, {
    t
  } = useTranslation(variantsLocaleNamespace), {
    createVariant
  } = useVariantOperations();
  let t0;
  $[0] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t0 = getVariantDefaults(), $[0] = t0) : t0 = $[0];
  const initialValue = t0;
  let t1;
  $[1] !== createVariant || $[2] !== onSubmit ? (t1 = async (variant) => {
    await createVariant(variant), onSubmit(variant._id);
  }, $[1] = createVariant, $[2] = onSubmit, $[3] = t1) : t1 = $[3];
  const handleSubmit = t1;
  let t2;
  $[4] !== t ? (t2 = t("dialog.create.action.confirm"), $[4] = t, $[5] = t2) : t2 = $[5];
  let t3;
  $[6] !== t ? (t3 = t("dialog.create.error.title"), $[6] = t, $[7] = t3) : t3 = $[7];
  let t4;
  $[8] !== t ? (t4 = t("dialog.create.title"), $[8] = t, $[9] = t4) : t4 = $[9];
  let t5;
  return $[10] !== handleSubmit || $[11] !== onCancel || $[12] !== t2 || $[13] !== t3 || $[14] !== t4 ? (t5 = /* @__PURE__ */ jsx(VariantDialog, { confirmDataTestId: "submit-variant-button", confirmText: t2, errorTitle: t3, header: t4, id: "create-variant-dialog", initialValue, onCancel, onSubmit: handleSubmit }), $[10] = handleSubmit, $[11] = onCancel, $[12] = t2, $[13] = t3, $[14] = t4, $[15] = t5) : t5 = $[15], t5;
}
function VariantMenuButton(t0) {
  const $ = c(20), {
    variant
  } = t0, {
    t
  } = useTranslation(variantsLocaleNamespace), toast = useToast(), {
    deleteVariant
  } = useVariantOperations(), [isDeleting, setIsDeleting] = useState(!1);
  let t1;
  $[0] !== deleteVariant || $[1] !== t || $[2] !== toast || $[3] !== variant._id ? (t1 = async () => {
    setIsDeleting(!0);
    try {
      await deleteVariant(variant._id), setIsDeleting(!1);
    } catch (t22) {
      console.error(t22), toast.push({
        closable: !0,
        status: "error",
        title: t("overview.action.delete-variant.error.title")
      }), setIsDeleting(!1);
    }
  }, $[0] = deleteVariant, $[1] = t, $[2] = toast, $[3] = variant._id, $[4] = t1) : t1 = $[4];
  const handleDelete = t1;
  let t2;
  $[5] !== isDeleting ? (t2 = /* @__PURE__ */ jsx(ContextMenuButton, { disabled: isDeleting, loading: isDeleting }), $[5] = isDeleting, $[6] = t2) : t2 = $[6];
  let t3;
  $[7] !== variant._id ? (t3 = getVariantId(variant._id), $[7] = variant._id, $[8] = t3) : t3 = $[8];
  const t4 = `variant-actions-${t3}`;
  let t5;
  $[9] !== t ? (t5 = t("overview.action.delete-variant"), $[9] = t, $[10] = t5) : t5 = $[10];
  let t6;
  $[11] !== handleDelete || $[12] !== isDeleting || $[13] !== t5 ? (t6 = /* @__PURE__ */ jsx(Menu, { children: /* @__PURE__ */ jsx(MenuItem, { disabled: isDeleting, onClick: handleDelete, text: t5, tone: "critical" }) }), $[11] = handleDelete, $[12] = isDeleting, $[13] = t5, $[14] = t6) : t6 = $[14];
  let t7;
  $[15] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t7 = {
    placement: "bottom-end",
    portal: !0
  }, $[15] = t7) : t7 = $[15];
  let t8;
  return $[16] !== t2 || $[17] !== t4 || $[18] !== t6 ? (t8 = /* @__PURE__ */ jsx(MenuButton, { button: t2, id: t4, menu: t6, popover: t7 }), $[16] = t2, $[17] = t4, $[18] = t6, $[19] = t8) : t8 = $[19], t8;
}
const BASE_PANEL = "M64 150.174V81.8262C64 79.1748 65.498 76.751 67.8695 75.5653L103.87 57.5652C108.524 55.2381 114 58.6226 114 63.8262V132.174C114 134.825 112.502 137.249 110.131 138.435L74.1305 156.435C69.4762 158.762 64 155.377 64 150.174Z", LINES_FULL = ["M72 86 L106 86", "M72 96 L102 96", "M72 106 L104 106", "M72 116 L98 116"], LINES_MEDIUM = ["M72 88 L100 88", "M72 98 L96 98", "M72 108 L102 108"], LINES_SHORT = ["M72 90 L94 90", "M72 100 L92 100"], VariantIllustration = () => {
  const $ = c(6);
  let t0;
  $[0] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t0 = /* @__PURE__ */ jsxs("g", { opacity: 0.12, children: [
    /* @__PURE__ */ jsx("path", { d: BASE_PANEL, fill: "var(--card-muted-bg-color)", stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2, transform: "translate(12 10)" }),
    /* @__PURE__ */ jsx("path", { d: BASE_PANEL, fill: "var(--card-muted-bg-color)", stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2, transform: "translate(6 6)" })
  ] }), $[0] = t0) : t0 = $[0];
  let t1;
  $[1] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t1 = /* @__PURE__ */ jsx("g", { opacity: 0.28, children: /* @__PURE__ */ jsxs("g", { transform: "translate(118 6) scale(0.5)", children: [
    /* @__PURE__ */ jsx("path", { d: BASE_PANEL, fill: "var(--card-muted-bg-color)", stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2 }),
    LINES_MEDIUM.map(_temp$1)
  ] }) }), $[1] = t1) : t1 = $[1];
  let t2;
  $[2] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t2 = /* @__PURE__ */ jsx("g", { opacity: 0.34, children: /* @__PURE__ */ jsxs("g", { transform: "translate(126 52) scale(0.52)", children: [
    /* @__PURE__ */ jsx("path", { d: BASE_PANEL, fill: "var(--card-muted-bg-color)", stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2 }),
    LINES_FULL.slice(0, 3).map(_temp2)
  ] }) }), $[2] = t2) : t2 = $[2];
  let t3;
  $[3] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t3 = /* @__PURE__ */ jsx("g", { opacity: 0.26, children: /* @__PURE__ */ jsxs("g", { transform: "translate(106 102) scale(0.48)", children: [
    /* @__PURE__ */ jsx("path", { d: BASE_PANEL, fill: "var(--card-muted-bg-color)", stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2 }),
    LINES_SHORT.map(_temp3)
  ] }) }), $[3] = t3) : t3 = $[3];
  let t4;
  $[4] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t4 = /* @__PURE__ */ jsxs("g", { opacity: 0.14, stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2, strokeLinecap: "round", children: [
    /* @__PURE__ */ jsx("path", { d: "M118 112 Q 152 72 168 58", fill: "none" }),
    /* @__PURE__ */ jsx("path", { d: "M124 124 Q 168 108 188 92", fill: "none" }),
    /* @__PURE__ */ jsx("path", { d: "M118 138 Q 150 152 162 168", fill: "none" })
  ] }), $[4] = t4) : t4 = $[4];
  let t5;
  return $[5] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t5 = /* @__PURE__ */ jsxs("svg", { "aria-hidden": "true", "data-testid": "variant-illustration", width: "248", height: "201", viewBox: "0 0 248 201", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
    t0,
    t1,
    t2,
    t3,
    t4,
    /* @__PURE__ */ jsxs("g", { transform: "translate(26 20) scale(1.06)", children: [
      /* @__PURE__ */ jsx("path", { d: BASE_PANEL, fill: "var(--card-muted-bg-color)", stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2, opacity: 0.92 }),
      LINES_FULL.map(_temp4)
    ] })
  ] }), $[5] = t5) : t5 = $[5], t5;
};
function _temp$1(d) {
  return /* @__PURE__ */ jsx("path", { d, stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2, strokeLinecap: "round", opacity: 0.55 }, d);
}
function _temp2(d_0) {
  return /* @__PURE__ */ jsx("path", { d: d_0, stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2, strokeLinecap: "round", opacity: 0.5 }, d_0);
}
function _temp3(d_1) {
  return /* @__PURE__ */ jsx("path", { d: d_1, stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2, strokeLinecap: "round", opacity: 0.55 }, d_1);
}
function _temp4(d_2, index) {
  return /* @__PURE__ */ jsx("path", { d: d_2, stroke: "var(--card-muted-fg-color)", strokeWidth: 1.2, strokeLinecap: "round", opacity: 0.42 + index * 0.06 }, d_2);
}
const VARIANTS_DOCUMENTATION_URL = "https://www.sanity.io/docs/content-variants", VariantsEmptyState = (t0) => {
  const $ = c(22), {
    createVariantButton
  } = t0, {
    t
  } = useTranslation(variantsLocaleNamespace);
  let t1;
  $[0] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t1 = {
    maxWidth: rem(300)
  }, $[0] = t1) : t1 = $[0];
  let t2;
  $[1] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t2 = /* @__PURE__ */ jsx(VariantIllustration, {}), $[1] = t2) : t2 = $[1];
  let t3;
  $[2] !== t ? (t3 = t("overview.title"), $[2] = t, $[3] = t3) : t3 = $[3];
  let t4;
  $[4] !== t3 ? (t4 = /* @__PURE__ */ jsx(Text, { as: "h1", size: 1, weight: "semibold", "data-testid": "no-variants-info-text", children: t3 }), $[4] = t3, $[5] = t4) : t4 = $[5];
  let t5;
  $[6] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t5 = {
    textAlign: "center"
  }, $[6] = t5) : t5 = $[6];
  let t6;
  $[7] !== t ? (t6 = t("overview.empty-state.description"), $[7] = t, $[8] = t6) : t6 = $[8];
  let t7;
  $[9] !== t6 ? (t7 = /* @__PURE__ */ jsx(Text, { size: 1, muted: !0, style: t5, children: t6 }), $[9] = t6, $[10] = t7) : t7 = $[10];
  let t8;
  $[11] !== t ? (t8 = t("overview.action.documentation"), $[11] = t, $[12] = t8) : t8 = $[12];
  let t9;
  $[13] !== t8 ? (t9 = /* @__PURE__ */ jsx(Button, { as: "a", href: VARIANTS_DOCUMENTATION_URL, target: "_blank", mode: "ghost", text: t8 }), $[13] = t8, $[14] = t9) : t9 = $[14];
  let t10;
  $[15] !== createVariantButton || $[16] !== t9 ? (t10 = /* @__PURE__ */ jsxs(Flex, { gap: 2, children: [
    createVariantButton,
    t9
  ] }), $[15] = createVariantButton, $[16] = t9, $[17] = t10) : t10 = $[17];
  let t11;
  return $[18] !== t10 || $[19] !== t4 || $[20] !== t7 ? (t11 = /* @__PURE__ */ jsxs(Flex, { align: "center", "data-testid": "variants-empty-state", flex: 1, direction: "column", gap: 3, style: t1, paddingBottom: 5, children: [
    t2,
    t4,
    t7,
    t10
  ] }), $[18] = t10, $[19] = t4, $[20] = t7, $[21] = t11) : t11 = $[21], t11;
}, VariantDocumentsCell = (t0) => {
  const $ = c(6), {
    cellProps,
    datum
  } = t0;
  if (datum.isLoading) {
    let t12;
    $[0] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t12 = /* @__PURE__ */ jsx(Text, { size: 1, children: /* @__PURE__ */ jsx(Skeleton, { animated: !0, radius: 1, style: {
      width: "4ch"
    } }) }), $[0] = t12) : t12 = $[0];
    let t22;
    return $[1] !== cellProps ? (t22 = /* @__PURE__ */ jsx(Flex, { ...cellProps, align: "center", paddingX: 2, paddingY: 3, sizing: "border", children: t12 }), $[1] = cellProps, $[2] = t22) : t22 = $[2], t22;
  }
  let t1;
  $[3] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t1 = /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: "0" }), $[3] = t1) : t1 = $[3];
  let t2;
  return $[4] !== cellProps ? (t2 = /* @__PURE__ */ jsx(Flex, { ...cellProps, align: "center", paddingX: 2, paddingY: 3, sizing: "border", children: t1 }), $[4] = cellProps, $[5] = t2) : t2 = $[5], t2;
}, VariantTitleCell = (t0) => {
  const $ = c(28), {
    cellProps,
    datum: variant
  } = t0, {
    t
  } = useTranslation(variantsLocaleNamespace);
  let t1;
  $[0] !== variant._id ? (t1 = getVariantId(variant._id), $[0] = variant._id, $[1] = t1) : t1 = $[1];
  const encodedVariantId = t1;
  let t2;
  $[2] !== encodedVariantId ? (t2 = forwardRef(function(linkProps, ref) {
    return /* @__PURE__ */ jsx(StateLink, { ...linkProps, ref, state: {
      variantId: encodedVariantId
    } });
  }), $[2] = encodedVariantId, $[3] = t2) : t2 = $[3];
  const VariantLink = t2;
  if (variant.isLoading) {
    let t32;
    $[4] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t32 = /* @__PURE__ */ jsx(Text, { size: 1, weight: "medium", children: /* @__PURE__ */ jsx(Skeleton, { animated: !0, radius: 1, style: {
      width: "16ch"
    } }) }), $[4] = t32) : t32 = $[4];
    let t42;
    $[5] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t42 = /* @__PURE__ */ jsxs(Stack, { space: 2, children: [
      t32,
      /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: /* @__PURE__ */ jsx(Skeleton, { animated: !0, radius: 1, style: {
        width: "32ch"
      } }) })
    ] }), $[5] = t42) : t42 = $[5];
    let t52;
    return $[6] !== cellProps ? (t52 = /* @__PURE__ */ jsx(Box, { ...cellProps, flex: 1, paddingLeft: 3, paddingRight: 2, paddingY: 2, sizing: "border", children: t42 }), $[6] = cellProps, $[7] = t52) : t52 = $[7], t52;
  }
  let t3;
  $[8] !== variant.conditions ? (t3 = getVariantConditionsText(variant.conditions), $[8] = variant.conditions, $[9] = t3) : t3 = $[9];
  const conditionsText = t3;
  let t4;
  $[10] !== variant ? (t4 = getVariantTitle(variant), $[10] = variant, $[11] = t4) : t4 = $[11];
  let t5;
  $[12] !== t4 ? (t5 = /* @__PURE__ */ jsx(Text, { size: 1, weight: "medium", children: t4 }), $[12] = t4, $[13] = t5) : t5 = $[13];
  let t6;
  $[14] !== conditionsText || $[15] !== t ? (t6 = conditionsText || t("overview.table.no-conditions"), $[14] = conditionsText, $[15] = t, $[16] = t6) : t6 = $[16];
  let t7;
  $[17] !== t6 ? (t7 = /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: t6 }), $[17] = t6, $[18] = t7) : t7 = $[18];
  let t8;
  $[19] !== t5 || $[20] !== t7 ? (t8 = /* @__PURE__ */ jsx(Flex, { align: "center", gap: 3, children: /* @__PURE__ */ jsxs(Stack, { flex: 1, space: 2, children: [
    t5,
    t7
  ] }) }), $[19] = t5, $[20] = t7, $[21] = t8) : t8 = $[21];
  let t9;
  $[22] !== VariantLink || $[23] !== t8 ? (t9 = /* @__PURE__ */ jsx(Card, { as: VariantLink, "data-as": "a", flex: 1, padding: 2, radius: 2, tone: "inherit", children: t8 }), $[22] = VariantLink, $[23] = t8, $[24] = t9) : t9 = $[24];
  let t10;
  return $[25] !== cellProps || $[26] !== t9 ? (t10 = /* @__PURE__ */ jsx(Box, { ...cellProps, flex: 1, paddingLeft: 3, paddingRight: 2, paddingY: 1, sizing: "border", children: t9 }), $[25] = cellProps, $[26] = t9, $[27] = t10) : t10 = $[27], t10;
};
function variantsOverviewColumnDefs(t) {
  return [{
    id: "metadata.title",
    sorting: !0,
    width: null,
    style: {
      minWidth: "min(50%, calc(100vw - 80px))"
    },
    header: (props) => /* @__PURE__ */ jsx(Flex, { ...props.headerProps, flex: 1, paddingLeft: 3, paddingRight: 2, paddingY: 3, sizing: "border", children: /* @__PURE__ */ jsx(Headers.SortHeaderButton, { ...props, text: t("overview.table.variant") }) }),
    cell: VariantTitleCell,
    sortTransform: (variant) => getVariantTitle(variant)
  }, {
    id: "documentCount",
    sorting: !1,
    width: 120,
    header: ({
      headerProps
    }) => /* @__PURE__ */ jsx(Flex, { ...headerProps, paddingY: 3, sizing: "border", children: /* @__PURE__ */ jsx(Headers.BasicHeader, { text: t("overview.table.documents") }) }),
    cell: VariantDocumentsCell
  }];
}
const VARIANT_TABLE_ROW_ID = "_id";
function VariantsOverview() {
  const $ = c(62), {
    t
  } = useTranslation(variantsLocaleNamespace), router = useRouter(), {
    data: variants,
    error,
    loading
  } = useAllVariants(), [scrollContainerRef, setScrollContainerRef] = useState(null), [searchQuery, setSearchQuery] = useState(""), [isCreateVariantDialogOpen, setIsCreateVariantDialogOpen] = useState(!1);
  let t0;
  $[0] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t0 = () => {
    setIsCreateVariantDialogOpen(!0);
  }, $[0] = t0) : t0 = $[0];
  const handleCreateVariant = t0;
  let t1;
  $[1] !== router ? (t1 = (createdVariantId) => {
    setIsCreateVariantDialogOpen(!1), router.navigate({
      variantId: getVariantId(createdVariantId)
    });
  }, $[1] = router, $[2] = t1) : t1 = $[2];
  const handleOnCreateVariant = t1;
  let t2;
  $[3] !== t ? (t2 = variantsOverviewColumnDefs(t), $[3] = t, $[4] = t2) : t2 = $[4];
  const columnDefs = t2, renderRowActions = _temp;
  let t3;
  $[5] !== variants ? (t3 = variants ?? [], $[5] = variants, $[6] = t3) : t3 = $[6];
  const variantsList = t3;
  let t4;
  $[7] !== searchQuery || $[8] !== variantsList ? (t4 = filterVariantsForSearch(variantsList, searchQuery), $[7] = searchQuery, $[8] = variantsList, $[9] = t4) : t4 = $[9];
  const filteredVariants = t4, hasVariants = variantsList.length > 0;
  let t5;
  $[10] !== t ? (t5 = t("overview.action.create-variant"), $[10] = t, $[11] = t5) : t5 = $[11];
  let t6;
  $[12] !== isCreateVariantDialogOpen || $[13] !== t5 ? (t6 = /* @__PURE__ */ jsx(Button, { disabled: isCreateVariantDialogOpen, icon: AddIcon, onClick: handleCreateVariant, text: t5 }), $[12] = isCreateVariantDialogOpen, $[13] = t5, $[14] = t6) : t6 = $[14];
  const createVariantButton = t6;
  let t7;
  $[15] !== createVariantButton || $[16] !== error || $[17] !== hasVariants || $[18] !== t ? (t7 = () => error && !hasVariants ? /* @__PURE__ */ jsx(Flex, { align: "center", direction: "column", gap: 3, justify: "center", padding: 4, children: /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: t("overview.error") }) }) : /* @__PURE__ */ jsx(VariantsEmptyState, { createVariantButton }), $[15] = createVariantButton, $[16] = error, $[17] = hasVariants, $[18] = t, $[19] = t7) : t7 = $[19];
  const tableEmptyState = t7;
  let t8;
  $[20] !== t ? (t8 = t("overview.title"), $[20] = t, $[21] = t8) : t8 = $[21];
  let t9;
  $[22] !== t8 ? (t9 = /* @__PURE__ */ jsx(Text, { as: "h1", size: 4, weight: "bold", children: t8 }), $[22] = t8, $[23] = t9) : t9 = $[23];
  let t10;
  $[24] !== t ? (t10 = t("overview.description"), $[24] = t, $[25] = t10) : t10 = $[25];
  let t11;
  $[26] !== t10 ? (t11 = /* @__PURE__ */ jsx(Text, { muted: !0, size: 1, children: t10 }), $[26] = t10, $[27] = t11) : t11 = $[27];
  let t12;
  $[28] !== t11 || $[29] !== t9 ? (t12 = /* @__PURE__ */ jsxs(Stack, { space: 3, children: [
    t9,
    t11
  ] }), $[28] = t11, $[29] = t9, $[30] = t12) : t12 = $[30];
  let t13;
  $[31] !== createVariantButton || $[32] !== t12 ? (t13 = /* @__PURE__ */ jsx(Card, { flex: "none", paddingY: 5, children: /* @__PURE__ */ jsxs(Flex, { align: "flex-start", gap: 4, justify: "space-between", children: [
    t12,
    createVariantButton
  ] }) }), $[31] = createVariantButton, $[32] = t12, $[33] = t13) : t13 = $[33];
  const t14 = !!searchQuery;
  let t15, t16;
  $[34] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t15 = (event) => setSearchQuery(event.currentTarget.value), t16 = () => setSearchQuery(""), $[34] = t15, $[35] = t16) : (t15 = $[34], t16 = $[35]);
  let t17;
  $[36] !== t ? (t17 = t("overview.search.placeholder"), $[36] = t, $[37] = t17) : t17 = $[37];
  let t18;
  $[38] !== searchQuery || $[39] !== t14 || $[40] !== t17 ? (t18 = /* @__PURE__ */ jsx(Box, { flex: "none", paddingBottom: 4, paddingTop: 2, children: /* @__PURE__ */ jsx(TextInput, { clearButton: t14, fontSize: 1, icon: SearchIcon, onChange: t15, onClear: t16, placeholder: t17, radius: 3, value: searchQuery }) }), $[38] = searchQuery, $[39] = t14, $[40] = t17, $[41] = t18) : t18 = $[41];
  let t19;
  $[42] !== error || $[43] !== t ? (t19 = error && /* @__PURE__ */ jsx(Card, { flex: "none", padding: 3, tone: "critical", children: /* @__PURE__ */ jsx(Text, { size: 1, children: t("overview.error") }) }), $[42] = error, $[43] = t, $[44] = t19) : t19 = $[44];
  let t20;
  $[45] !== t13 || $[46] !== t18 || $[47] !== t19 ? (t20 = /* @__PURE__ */ jsx(Container, { flex: "none", width: 3, children: /* @__PURE__ */ jsxs(Flex, { direction: "column", paddingX: 3, children: [
    t13,
    t18,
    t19
  ] }) }), $[45] = t13, $[46] = t18, $[47] = t19, $[48] = t20) : t20 = $[48];
  let t21;
  $[49] !== columnDefs || $[50] !== filteredVariants || $[51] !== loading || $[52] !== scrollContainerRef || $[53] !== tableEmptyState ? (t21 = /* @__PURE__ */ jsx(Box, { flex: 1, overflow: "auto", ref: setScrollContainerRef, children: /* @__PURE__ */ jsx(Table, { columnDefs, data: filteredVariants, emptyState: tableEmptyState, loading, rowId: VARIANT_TABLE_ROW_ID, rowActions: renderRowActions, scrollContainerRef }) }), $[49] = columnDefs, $[50] = filteredVariants, $[51] = loading, $[52] = scrollContainerRef, $[53] = tableEmptyState, $[54] = t21) : t21 = $[54];
  let t22;
  $[55] !== handleOnCreateVariant || $[56] !== isCreateVariantDialogOpen ? (t22 = isCreateVariantDialogOpen && /* @__PURE__ */ jsx(CreateVariantDialog, { onCancel: () => setIsCreateVariantDialogOpen(!1), onSubmit: handleOnCreateVariant }), $[55] = handleOnCreateVariant, $[56] = isCreateVariantDialogOpen, $[57] = t22) : t22 = $[57];
  let t23;
  return $[58] !== t20 || $[59] !== t21 || $[60] !== t22 ? (t23 = /* @__PURE__ */ jsxs(Flex, { direction: "column", flex: 1, height: "fill", children: [
    t20,
    t21,
    t22
  ] }), $[58] = t20, $[59] = t21, $[60] = t22, $[61] = t23) : t23 = $[61], t23;
}
function _temp(t0) {
  const {
    datum
  } = t0;
  return /* @__PURE__ */ jsx(VariantMenuButton, { variant: datum });
}
function VariantsTool() {
  const $ = c(3), router = useRouter(), variantId = typeof router.state.variantId == "string" ? router.state.variantId : void 0;
  if (variantId) {
    let t02;
    return $[0] !== variantId ? (t02 = /* @__PURE__ */ jsx(VariantDetail, {}, variantId), $[0] = variantId, $[1] = t02) : t02 = $[1], t02;
  }
  let t0;
  return $[2] === /* @__PURE__ */ Symbol.for("react.memo_cache_sentinel") ? (t0 = /* @__PURE__ */ jsx(VariantsOverview, {}), $[2] = t0) : t0 = $[2], t0;
}
export {
  VariantsTool
};
